index.js

```diff
@@ -1,18 +1,151 @@
+    .toLowerCase();
+  return string;
+}
+
+// Creates a random integer between two numbers as efficiently as possible, for computer guessing
+function binarySearchAlgorithm(min, max) {
+  return Math.floor((max - min) / 2 + min);
+}
+
+// Computer uses this to randomly guess a number in range
+function randomInteger(min, max) {
+  min = Math.ceil(min);
+  max = Math.floor(max);
+  return Math.floor(Math.random() * (max - min + 1)) + min;
+}
+
+// Starts the game
+chooseGameType();
+
+//Sets up if computer guesses or player guesses
+async function chooseGameType() {
+  console.log(`\nPlease type one of these to choose a game type: ${validGameTypeInput.join(', ')}.\n`);
+
+  // Sets up chooseGameType to be checked if valid input
+  let gameTypeInput = '';
+
+  // Sanitizes user input to choose game type
+  while (!validGameTypeInput.includes(sanitizeString(gameTypeInput))) {
+    // User input enables game type
+    gameTypeInput = await ask('Would you (player) like to guess the number or have me (computer) guess the number? ');
+
+    // Chooses which game type is initiated
+    if (playComputerGuessing.includes(sanitizeString(gameTypeInput))) {
+      computerStart(); // Chooses computer will guess
+    }
+
+    if (playPlayerGuessing.includes(sanitizeString(gameTypeInput))) {
+      playerStart(); // Chooses player will guess
+    }
+  }
+}
+
+// Computer guesses your number ------------------------------------
+async function computerStart() {
+  console.log("\nLet's play a game where you (human) make up a number\nand I (computer) try to guess it.\n");
+
+  // Game setup ----------------------------------------------------
+  setRangeMin = +setRangeMin; // changes from object to a number
+  setRangeMax = +setRangeMax;
+  let guessCounter = 0;
+
+  const secretNumber = await ask('What is your secret number? ');
 
-async function start() {
-  console.log("Let's play a game where you (human) make up a number and I (computer) try to guess it.")
-  let secretNumber = await ask("What is your secret number?\nI won't peek, I promise...\n");
-  console.log('You entered: ' + secretNumber);
-  // Now try and complete the program.
+  // Initializes first random number guessed
+  let computerGuess = binarySearchAlgorithm(setRangeMin, setRangeMax);
+
+  // Sets up guessQuestion to include correct answer
+  let guessQuestion = '';
+
+  // Game logic ----------------------------------------------------
+  while (!correctAnswer.includes(sanitizeString(guessQuestion))) {
+    guessQuestion = await ask(`Is your number ${computerGuess}? `);
+    guessCounter += 1;
+
+    if (wrongAnswer.includes(sanitizeString(guessQuestion))) {
+      let highOrLow = await ask('Is it higher or lower? ');
+
+      // Start HighOrLow if loop branch
+      if (highAnswer.includes(sanitizeString(highOrLow))) {
+
+        console.log('Looks like I need to guess higher.');
+        setRangeMin = computerGuess + 1;
+        computerGuess = randomInteger(setRangeMin, setRangeMax);
```

Comment:
Did you mean to use your binarySearchAlgorithm function here and line 108 so that it keeps guessing using the binary search method?